tests/manage/test_add_mds_to_cluster.py

```diff
@@ -2,92 +2,58 @@
 A test for creating a CephFS
 """
 import logging
-
 import pytest
 
 from ocs_ci.ocs import constants, defaults, ocp
-from ocs_ci.framework import config
 from ocs_ci.framework.testlib import tier1, ManageTest
 from ocs_ci.ocs.resources.ocs import OCS
 
+from tests import helpers
+
 log = logging.getLogger(__name__)
 
 CEPHFS_DELETED = '"{cephfs_name}" deleted'
 
-POD = ocp.OCP(kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace'])
-CEPHFS = ocp.OCP(kind=constants.CEPHFILESYSTEM)
-CEPH_OBJ = None
 
+@pytest.fixture(scope="class")
+def cephfs():
+    cephfs = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+    return cephfs
 
-@pytest.fixture(scope='class')
-def test_fixture(request):
-    """
-    Create disks
-    """
-    self = request.node.cls
 
-    def finalizer():
-        teardown(self)
-    request.addfinalizer(finalizer)
-    setup(self)
-
-
-def setup(self):
-    """
-    Setting up the environment for the test
-    """
-    global CEPHFS, CEPH_OBJ
-    CEPHFS = ocp.OCP(kind=constants.CEPHFILESYSTEM)
-    self.fs_data = CEPHFS.get(defaults.CEPHFILESYSTEM_NAME)
-    self.fs_name = self.fs_data['metadata']['name']
-    CEPH_OBJ = OCS(**self.fs_data)
+@pytest.fixture(scope="class")
+def fs_data(cephfs):
+    fs_data = cephfs.get(defaults.CEPHFILESYSTEM_NAME)
+    return fs_data
 
 
-def teardown(self):
-    """
-    Tearing down the environment
-    """
-    self.fs_data = CEPHFS.get(defaults.CEPHFILESYSTEM_NAME)
-    self.fs_data['spec']['metadataServer']['activeCount'] = (
-        self.original_active_count
-    )
-    CEPH_OBJ.apply(**self.fs_data)
-    assert verify_fs_exist(self.original_active_count * 2)
+@pytest.fixture(scope="class")
+def ceph_obj(request, fs_data):
+    ceph_obj = OCS(**fs_data)
```

Comment:
And if I want 1 more ceph object? a different one

Reply:
If we wanted a fixture to be able to provide any number of resources we could do what @andrewschoen suggested in yesterdays meeting and have the fixture return a function that you can call to create the object instead of returning the object directly.

Reply:
Yeah, fixtures can be factory methods as well. Here's some good documentation on that. https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures